app/controllers/subscriptions_controller.rb

```diff
@@ -2,31 +2,36 @@ class SubscriptionsController < ApplicationController
   def create
     return render json: { id: 0 }, status: :created if smoke_test_address?
 
-    existing_subscription = nil
-    subscription = nil
+    subscription = send_email = status = nil
 
     Subscription.transaction do
+      deactivated_subscriber = subscriber.deactivated?
+      subscriber.activate! if deactivated_subscriber
+
       existing_subscription = Subscription.active.lock.find_by(
         subscriber: subscriber,
         subscriber_list: subscriber_list,
       )
 
-      existing_subscription.end(reason: :frequency_changed) if existing_subscription
+      create_new_subscription = existing_subscription.nil? ||
+        (frequency.to_s != existing_subscription.frequency.to_s)
 
-      subscriber.activate! if subscriber.deactivated?
+      if create_new_subscription
+        existing_subscription&.end(reason: :frequency_changed)
+        new_subscription = Subscription.create!(
+          subscriber: subscriber,
+          subscriber_list: subscriber_list,
+          frequency: frequency,
+          signon_user_uid: current_user.uid,
+          source: existing_subscription ? :frequency_changed : :user_signed_up,
+          )
+      end
 
-      subscription = Subscription.create!(
-        subscriber: subscriber,
-        subscriber_list: subscriber_list,
-        frequency: frequency,
-        signon_user_uid: current_user.uid,
-        source: existing_subscription ? :frequency_changed : :user_signed_up,
-      )
+      subscription = new_subscription || existing_subscription
+      send_email = create_new_subscription || deactivated_subscriber
```

Comment:
Minor, but `deactivated_subscriber` is strange to read down here. How about changing the assignment to be

```
if subscriber.deactivated?
   subscriber_was_deactivated = true
   subscriber.activate!
end
```

It's not great, but it does mean the subsequent code is easier to understand.

Reply:
updated it. let me know what you think!